app/services/conversation_service.rb

```diff
@@ -0,0 +1,17 @@
+class ConversationService
+  def self.create_chat(current_user, matched_user)
+    return if already_created?(current_user, matched_user)
+
+    conversation = current_user.conversations.create!(user_id: current_user.id)
+    matched_user.conversations << conversation
+    conversation
+  end
+
+  def self.already_created?(current_user, matched_user)
```

Comment:
I think you can avoid the usage of `result` variable here by doing this:
```
def self.already_created?(current_user, matched_user)
  current_user.conversations.any? { |conv| conv.users.include?(matched_user) }
end

```

`.any?` will return false if there are not conversations for the current user and if there is one or more, if will continue to evaluate this line `conv.users.include?(matched_user)`

Reply:
Same as I mention to the other Rodri 😄 
you could rename `conv` to `conversation` to make it clearer that the variable refers to the singular

Reply:
Done! Thanks for the tip.

Reply:
Done! Thanks for the comment.